net-analyzer/ossec-hids/ossec-hids-3.1.0.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit user
+
+DESCRIPTION="Open Source Host-based Intrusion Detection System"
+HOMEPAGE="https://www.ossec.net/"
+SRC_URI="https://github.com/ossec/ossec-hids/archive/${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="GPL-2"
+SLOT="0"
+
+KEYWORDS="~amd64"
+
+IUSE="agent hybrid local mysql postgres server sqlite zlib"
+DEPEND="mysql? ( virtual/mysql )
+	sqlite? ( dev-db/sqlite:3 )
+	postgres? ( dev-db/postgresql:= )
+	zlib? ( sys-libs/zlib:= )"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/${P}/src"
+
+MY_PREFIX="/var/ossec"
+
+my_options() {
```

Comment:
Is there really a need to do this multiple times? Why not just process it in `src_configure`, and leave in some global variable for other phases to use.

Reply:
I actually wondered how to best avoid global variables, so I thought I'd use a function instead. I can of course change that, although I don't expect performance issues. What do you prefer?

Reply:
Well, given that an associative array needs to be converted to a string before passing the options, I think I'll stick with the existing function.

Reply:
Not associative array but a regular array; and no, it doesn't get 'converted to string'. it's passed as a list, preserving parameter boundaries. Unlike your 'string' that goes through word splitting and filename expansion.

Reply:
Sorry, but I don't understand what exactly you are suggesting. Is there an ebuild I could use to see how this method is used?

Reply:
```
src_configure() {
  FOO=(
    A=1
    B=2
    C="something with spaces here"
  )
  use d && FOO+=( D=1 )
}

src_compile() {
  emake "${FOO[@]}"
}

Reply:
```
src_configure() {
  FOO=(
    A=1
    B=2
    C="something with spaces here"
  )
  use d && FOO+=( D=1 )
}

src_compile() {
  emake "${FOO[@]}"
}

Reply:
```
src_configure() {
  FOO=(
    A=1
    B=2
    C="something with spaces here"
  )
  use d && FOO+=( D=1 )
}

src_compile() {
  emake "${FOO[@]}"
}
```

Reply:
Ah, key-value-pairs as array elements; I misunderstood your intention. I have now modified the ebuild to use a global array MY_OPT in the manner you described.